### ISSUE TITLE
Provision for Optional Cancellation Token Parameter for KeysetPaginateAsync and OffsetPaginateAsync

### ISSUE THREAD

**Issue by ugumerie** [mrahhal/MR.AspNetCore.Pagination#8]
I think there should be provision for passing down the optional cancellation token argument to these pagination methods most especially the `KeysetPaginateAsync` just like in EF Core linq extension methods. For example while implementing a feature like infinite scroll on mobile or web apps, I want cancel request as the user is scrolling down fast or has moved away from the screen.

Thanks
Regards

**Comment by mrahhal** [comment]
Good idea. I'll add this as up-for-grabs for now. If no one takes it I'll probably do it at some point.
